### PR TITLE
[GET /user/check-nickname] Verify Nickname

### LINKED ISSUE #9: [User API] Verify Nickname
See API docs for requirement / need to add test cases

### PR DESCRIPTION
**Pull request by hse2527**
Close #9 

## When Reviewing this PR, please
- checkout `feature-#9-steve` branch
- Thoroughly read the code and compare it with API documentation
- Thoroughly read the test code
- Check newly implemented datatypes
- Run the test code. Check the behavior and code coverage

### CODE REVIEW

File: src/datatypes/User/User.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    this.graduationYear = graduationYear;
+    this.tncVersion = tncVersion;
+  }
+
+  /**
+   * Retrieve nickname of the user
+   *
+   * @param {Cosmos.Database} dbClient DB Client (Cosmos Database)
+   */
+  static async readNicknames(dbClient: Cosmos.Database): Promise<string[]> {
```
Comment (hyecheol123) [inline src/datatypes/User/User.ts]: No need to read all Nicknames
- Just check existence of the nickname

File: src/functions/JWT/createAccessToken.ts
Comment (hyecheol123) [inline src/functions/JWT/createAccessToken.ts:1]: Token only be created by Auth API

Should not use this code here.

File: src/routes/user.ts [hunk 2]
```diff
@@ -44,9 +51,49 @@ const userRouter = express.Router();
...
+    }
+    verifyAccessToken(accessToken, req.app.get('jwtAccessKey'));
+
+    // Check request body
+    const nicknameVerifyRequest: { nickname: string } = req.body;
+    if (!validateVerifyNicknameRequest(nicknameVerifyRequest)) {
+      throw new BadRequestError();
+    }
+
+    // DB operation - check nickname availability
```
Comment (hyecheol123) [inline src/routes/user.ts:165]: Fix DB Operation / See the comments above

File: test/testCases/user/get.check-nickname.test.ts
Comment (hyecheol123) [inline test/testCases/user/get.check-nickname.test.ts:1]: As logic is wrong, this entire test code should be fixed/reviewed.

File: src/datatypes/User/User.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+   * @param {Date | string} lastLogin - last login date of the user
+   * @param {Date | string} signUpDate - sign up date of the user
+   * @param {Date | string} nicknameChanged - nickname changed date of the user
+   * @param {boolean} deleted - whether the user is deleted or not
+   * @param {boolean} locked - whether the user is locked or not
+   * @param {string} major - major of the user
+   * @param {number} graduationYear - graduation year of the user
+   * @param {string} tncVersion - terms and conditions version of the user
+   */
+  constructor(
```
Comment (hyecheol123) [inline src/datatypes/User/User.ts:43]: Use Default Value or Optional Parameter to handle deleted and locked user's additional information.
Need to validate input.

**Review by hyecheol123 — CHANGES_REQUESTED** [review]
Major Revision Requested

**Comment by hse2527** [comment]
Branch mistaken

**Pull request closed without merging**